lib/Dialect/Comb/CombFolds.cpp

```diff
@@ -988,6 +988,155 @@ OpFoldResult ICmpOp::fold(ArrayRef<Attribute> constants) {
   return {};
 }
 
+// Given a range of operands, computes the number of matching prefix and
+// suffix elements. This does not perform cross-element matching.
+template <typename Range>
+static size_t computeCommonPrefixLength(const Range &a, const Range &b) {
+  size_t commonPrefixLength = 0;
+  auto ia = a.begin();
+  auto ib = b.begin();
+
+  for (; ia != a.end() && ib != b.end(); ia++, ib++, commonPrefixLength++) {
+    if (*ia != *ib) {
+      break;
+    }
+  }
+
+  return commonPrefixLength;
+}
+
+static size_t getTotalWidth(const OperandRange &range) {
+  size_t totalWidth = 0;
+  for (auto operand : range) {
+    // getIntOrFloatBitWidth should never raise, since all arguments to ConcatOp
+    // are integers.
+    ssize_t width = operand.getType().getIntOrFloatBitWidth();
+    assert(width >= 0);
+    totalWidth += width;
+  }
+  return totalWidth;
+}
+
+static bool isAnyZeroWidthOperands(OperandRange operands) {
```

Comment:
Nit but I'd rename `isAnyZeroWidthOperands` -> hasAnyZeroWidthOperands.

Also please pass OperandRange by const& for consistency